### PR TITLE
Manchester syntax pulled out of main parser

### PR DESCRIPTION
**Pull request by inaimathi**
- New file named expression.clj containing the Manchester parser
  (the intent is that there be multiple expression parsers, not just
  manchester, even though it's currently the only one)
- Add string generation from manchester syntax
- Add some minimal manual tests for parsing and formatting (now that
  we've got a round-trip, it seems like it should be relatively easy to
  generate a lot of these tests. Generating the tree of manchester
  expressions will be closely related to the parser. Wonder if there's a
  way to build a generator based on the parser?)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -219,7 +218,7 @@ If a HOWL stream starts with blank lines, these will be parsed as blank blocks.
 
 This blank block:
 
-
```
Comment (jamesaoverton) [inline README.md:222]: I want to keep these spaces.

**Comment by jamesaoverton** [comment]
Good. Can we move `convert-expression` and related functions out of `nquads.clj`?

File: src/howl/expression.clj [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    SPACE = #'\\s+'"))
+
+(defn string-to-parse [string]
+  [:MANCHESTER_EXPRESSION (manchester-parser string)])
+
+(defn manchester-format [mn-tree]
+  (if (string? mn-tree)
+    mn-tree
+    (case (first mn-tree)
+      (:CLASS_EXPRESSION
```
Comment (jamesaoverton) [inline src/howl/expression.clj:37]: I didn't know about using case with multiple constants like this. Cool!

File: src/howl/expression.clj [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+       :DISJUNCTION :CONJUNCTION :NEGATION
+       :SOME :ONLY
+       :OBJECT_PROPERTY_EXPRESSION) (apply str (map manchester-format (rest mn-tree)))
+
+      :NAME (manchester-format (second mn-tree))
+      (:SPACE :LABEL) (second mn-tree)
+      :QUOTED_LABEL (apply str (rest mn-tree)))))
+
+(defn parse-to-string [expression-parse]
+  (let [[exp-type parse-tree] expression-parse]
```
Comment (jamesaoverton) [inline src/howl/expression.clj]: You can do this `let` binding in the argument vector. I'd do it that way, and make the arguments clear in a docstring.

**Comment by jamesaoverton** [comment]
Please include a docstring for every function that's more than a line or two.

File: README.md (hunk 1 above)
Comment (inaimathi) [reply README.md:222]: Grr. That's an `after-save` hook I've got on my local that deletes trailing whitespace. I'll put a special case in it for this file (and revert this before merging, obviously).

**Pull request merged**